xmake/modules/lib/detect/find_package.lua

```diff
@@ -83,9 +85,15 @@ function main(name, opt)
     -- trace
     if opt.verbose or option.get("verbose") then
         if result then
-            cprint("checking for the %s ... ${color.success}%s", name, result.version and result.version or "${text.success}")
+            -- only display manager of found package if the package we searched for
+            -- did not specify a package manager
+            local original_manager_name, _, _ = split_package_name(name)
+            local display_name = original_manager_name and package_name
+                or (manager_name .. "::" .. package_name)
```

Comment:
we need only the returned real manager name and package name. So we need not call split_package_name()

We need to ensure that the returned manager_name must be a non-empty string.

```lua
local display_name = (manager_name or "") .. "::" .. package_name
```

xmake l find_package luajit          -- manager_name is vcpkg/xmake/xx  or system (default)
xmake l find_package vcpkg::luajit  -- manager_name vcpkg

Reply:
Therefore, we don’t need to put split_package_name separately in the independent module file, just use it in package/manager/find_package.lua

Reply:
Oh yeah, sorry about no nil guard there.

Re the second point, to satisfy https://github.com/xmake-io/xmake/pull/954#issuecomment-693410906 I believe we still need to check if the requested manager is the same as the found manager. We can return the requested manager from `find_package` as an additional return value though, so there's no code duplication or extra file.